### PR TITLE
working

### PR DESCRIPTION
**Pull request by raymondmay95**

### CODE REVIEW

File: react-app/src/App.js [hunk 1]
```diff
@@ -21,59 +22,79 @@ function App() {
...
-          <User />
-        </ProtectedRoute>
-        {/* Feel free to delete this route if it doesn't work with you database. It was just for testing the waveform player */}
-        <ProtectedRoute exact path='/audioPlayerTest' authenticated={authenticated}>
-          <Waveform songId={12} canvasHeight={200} canvasWidth={1000} />
-          <WaveFormControls songId={12} />
-
-          <Waveform songId={11} canvasHeight={200} canvasWidth={1000} />
-          <WaveFormControls songId={11} />
+          <Route path="/songs/:songId">
```
Comment (raymondmay95) [inline react-app/src/App.js:45]: here is a new route that moved everything else down

File: react-app/src/App.js [hunk 2]
```diff
@@ -21,59 +22,79 @@ function App() {
...
+            exact
+            path="/audioPlayerTest"
+            authenticated={authenticated}
+          >
+            <Waveform songId={12} canvasHeight={200} canvasWidth={1000} />
+            <WaveFormControls songId={12} />
 
-          <Waveform songId={8} canvasHeight={200} canvasWidth={1000} />
-          <WaveFormControls songId={8} />
+            <Waveform songId={11} canvasHeight={200} canvasWidth={1000} />
```
Comment (raymondmay95) [inline react-app/src/App.js:74]: ask jamie about this songId

File: react-app/src/components/waveform/index.js [hunk 3]
```diff
@@ -1,165 +1,159 @@
...
-import { useDispatch, useSelector } from 'react-redux'
-import { setCheckpoint, setActiveSongData } from '../../store/song'
+import React, { useEffect, useState, useRef } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { setCheckpoint, setActiveSongData } from "../../store/song";
 // import classes from './Waveform.module.css'
 
-
 const Waveform = ({ songId, canvasWidth, canvasHeight }) => {
+  const [waveformData, setWaveFormData] = useState();
```
Comment (raymondmay95) [inline react-app/src/components/waveform/index.js:7]: looks like more white space

File: react-app/src/components/waveform/index.js [hunk 4]
```diff
@@ -1,165 +1,159 @@
...
-  let canvasRef = useRef()
-
+  let canvasRef = useRef();
 
   // Setting the initial data
   useEffect(() => {
     (async () => {
-      let response = await fetch(`/api/song/${songId}`)
-      let data = await response.json()
+      let response = await fetch(`/api/songs/${songId}`);
```
Comment (raymondmay95) [inline react-app/src/components/waveform/index.js:27]: my error from earlier. Backend route to get a song is /api/songs/:song_id

File: react-app/src/components/waveform/index.js [hunk 5]
```diff
@@ -1,165 +1,159 @@
...
+        return;
       }
 
       // If there is discrepency between the number of bars calculated in this function, and the number stored it the state, update the state
       // This acts as a throttle. The canvas is repainted in a useEffect that has numWaveBars as a dependency
       if (numBars && numBars !== numWaveformBars) {
-        setNumWaveformBars(numBars)
+        setNumWaveformBars(numBars);
       }
-
```
Comment (raymondmay95) [inline react-app/src/components/waveform/index.js:73]: white space

File: react-app/src/components/waveform/index.js [hunk 6]
```diff
@@ -168,32 +162,36 @@ const Waveform = ({ songId, canvasWidth, canvasHeight }) => {
...
         adjustedValue * breakPointRatio
-      )
-    })
-
-
-  }, [waveformData, numWaveformBars, numHighlightedBars, canvasHeight, canvasWidth])
+      );
+    });
+  }, [
+    waveformData,
```
Comment (raymondmay95) [inline react-app/src/components/waveform/index.js:168]: formatter

**Comment by alex-pezzati** [comment]
Going to close this merge to request another one. Hoping to fix the package-lock.json issues.

**Pull request closed without merging**